src/server/client.ts

```diff
@@ -393,6 +345,33 @@ export class Auth0Client {
     }
   }
 
+  private async setSessionStore(data:SessionData, req?: PagesRouterRequest | NextRequest, res?: PagesRouterResponse | NextResponse){
+    if (req && res) {
+      if (req instanceof NextRequest && res instanceof NextResponse) {
+        await this.sessionStore.set(req.cookies, res.cookies, data)
+      } else {
+        const resHeaders = new Headers()
+        const resCookies = new ResponseCookies(resHeaders)
+        const reqCookies = this.createRequestCookies(req as PagesRouterRequest)
+        await this.sessionStore.set(reqCookies, resCookies, data)
+        for (const [key, value] of resHeaders.entries()) {
+          (res as PagesRouterResponse).setHeader(key, value)
+        }
+      }
+    } else {
+      // app router usage: Server Components, Server Actions, Route Handlers
+      try {
+        await this.sessionStore.set(await cookies(), await cookies(), data)
+      } catch (e) {
+        if (process.env.NODE_ENV === "development") {
+          console.warn(
+            "Failed to persist the updated token set. `getAccessToken()` was likely called from a Server Component which cannot set cookies."
+          )
+        }
+      }
+    }
+  }
```

Comment:
Let's ensure we test that this works fine in all cases (Server Component, Pages, App, ...), both for access tokens and federated connection access tokens.